File: 2751-3000/3010. Divide an Array Into Subarrays With Minimum Cost I/solution.py
class Solution:
    def minimumCost(self, nums: List[int]) -> int:
        a = nums[1] 
        b = nums[2] 
        for i in range(3,len(nums)) : 
            if nums[i] < a : 
                b = min( b , a )
                a = nums[i]
            elif nums[i] < b : 
                b = nums[i]
        ans = nums[0] + a + b 
        return ans  

File: 2751-3000/3010. Divide an Array Into Subarrays With Minimum Cost I/solution.cpp
class Solution {
public:
    int minimumCost(vector<int>& nums) {
        int n = nums.size() ; 
        int a = nums[1] ; 
        int b = nums[2] ; 
        for( int i=3 ; i<n ; ++i ){
            if( nums[i] < a ){
                b = min( b , a ) ; 
                a = nums[i] ; 
            }else if( nums[i] < b ){
                b = nums[i] ; 
            }
        }
        return nums[0] + a + b ; 
    }
};